Automattic/wp-calypso
# Fix conditions for ecommerce trial plan

For users on the ecommerce trial plan we were showing theme badges related to the business plan instead of the ecommerce plan.

| Before  | After |
| ------------- | ------------- |
|<img width="1169" alt="Screenshot 2024-09-19 at 14 00 53" src="https://github.com/user-attachments/assets/222cd13e-579f-475a-8b06-5331611e4263">  | <img width="1183" alt="Screenshot 2024-09-19 at 14 00 39" src="https://github.com/user-attachments/assets/7cad2037-e051-4093-8b61-5b0075a9eab1"> |
|  <img width="500" alt="Screenshot 2024-09-17 at 13 37 19" src="https://github.com/user-attachments/assets/fd17786c-4d37-4d37-bd79-8e9cb6046e6b"> | <img width="485" alt="Screenshot 2024-09-17 at 13 38 18" src="https://github.com/user-attachments/assets/e27b69f6-2bb7-4f6a-8df9-0a4e820fc50a">  |
| <img width="521" alt="Screenshot 2024-09-17 at 13 36 59" src="https://github.com/user-attachments/assets/d71854a3-63a1-4a29-ac98-999aa55677e2"> | <img width="601" alt="Screenshot 2024-09-17 at 13 36 16" src="https://github.com/user-attachments/assets/854dfdce-ec38-4b2e-b1d5-00414ce7b461"> |

Related to #
https://github.com/Automattic/wp-calypso/issues/94541

## Proposed Changes

* Add checks in relevant code to show ecommerce or business plan info on the theme badges

## Why are these changes being made?


* Users on the ecommerce trial plan should be lead to update to the full plan instead of the business plan

## Testing Instructions
- Go to [wordpress.com/ecommerce](https://wordpress.com/ecommerce/)
- Press Get Started and proceed all the way through the flow
- Go to wordpress.com/themes/:siteSlug
- Look for themes that require update and hover on "Upgrade and Subscribe". The tooltip should show ecommerce link leading to the checkout page with the ecommerce plan in the cart.
- Search for "Construction Firm" in the theme filter field.
- Check the "Upgrade" badge on community themes. The tooltip should show ecommerce link leading to the checkout page with the ecommerce plan in the cart.

**Testing Theme Details**
- Visit: /theme/kubrick-2024/:siteSlug
- The primary button should say: "Upgrade to activate"
- When clicking it you should land to checkout with the ecommerce plan in the cart
- For a site that is not on the ecommerce trial the checkout item should be the business plan

- Finally test for regressions.



## Pre-merge Checklist
